.github/workflows/go-ci.yml

```diff
@@ -12,8 +11,31 @@ on:
       - main
 
 jobs:
-  test:
+  fmt: # New job for gofmt
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go environment
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+
+    - name: Check Go formatting
```

Comment:
I think this is simpler.

- name: Check go formatting
   run: test -z $(find . -name "*.go" | xargs gofmt -s -l)